### PR TITLE
chore: make RemoveEnvFromApp truly idempotent

### PR DESCRIPTION
**Pull request by bvtujo**
Makes RemoveEnvFromApp really idempotent; previously we were getting errors when no resources existed before trying to empty the s3 buckets and ecr repos. 

Also fixes a bug where the s3 bucket emptier doesn't get created in the right region and therefore can't empty the bucket.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

**Review by Lou1415926 — APPROVED** [review]
lgtm! If the code with the problems aren't released yet, I think we can keep the title as `chore`

Edit: I reviewed again after the idempotent commit, and the approval still hold 👍🏼

**Review by huanjani — APPROVED** [review]
LGTM

**Pull request merged**
